docs/binlog_http_api.md

```diff
@@ -0,0 +1,186 @@
+# TiDB-Binlog HTTP API
+
+## Pump
+
+`PumpIP` is the ip of the Pump server. `8250` is the default port of Pump.
+
+1. Get the current status of Pump
+
+    ```shell
+    curl http://{PumpIP}:8250/status
+    ```
+
+    ```shell
+    $curl http://127.0.0.1:8250/status
+
+   {
+       "status":{
+           "ip-172-16-5-71:8250":{
+               "nodeId":"ip-172-16-5-71:8250",
+               "host":"172.16.5.71:8250",
+               "state":"online",
+               "isAlive":false,
+               "score":0,
+               "label":null,
+               "maxCommitTS":412518831548007863,
+               "updateTS":412518831338292471
+           },
+           "ip-172-16-5-74:8250":{
+               "nodeId":"ip-172-16-5-74:8250",
+               "host":"172.16.5.74:8250",
+               "state":"online",
+               "isAlive":false,
+               "score":0,
+               "label":null,
+               "maxCommitTS":412518831548007869,
+               "updateTS":412518831325184860
+           },
+           "ip-172-16-5-75:8250":{
+               "nodeId":"ip-172-16-5-75:8250",
+               "host":"172.16.5.75:8250",
+               "state":"online",
+               "isAlive":false,
+               "score":0,
+               "label":null,
+               "maxCommitTS":412518831548007875,
+               "updateTS":412518831534899704
+           }
+       },
+       "CommitTS":412518831849473230,
+       "Checkpoint":{
+   
+       },
+       "ErrMsg":""
+   }
+    ```
+
+1. Get all metrics of Pump
+
+    ```shell
+    curl http://{PumpIP}:8250/metrics
+    ```
+
+1. Get the status of all drainers
+
+    ```shell
+    curl http://{PumpIP}:8250/drainers
+    ```
+
+    ```shell
+    $curl http://127.0.0.1:8250/drainers
+
+   [
+       {
+           "nodeId":"ip-172-16-5-70:8249",
+           "host":"172.16.5.70:8249",
+           "state":"paused",
+           "isAlive":false,
+           "score":0,
+           "label":null,
+           "maxCommitTS":412361808537191540,
+           "updateTS":412518474704487076
+       }
+   ]
+    ```
+1. Change the Pump status
+
+    `NodeID` is the node id of the Pump server. `Action` is the action to execute[possible values: `pause`, `close`].
+
+    ```shell
+    curl -X PUT http://{PumpIP}:8250/state/{NodeID}/{Action}
```

Comment:
Question here:  in binlogctl we have `pause-pump`, `offline-pump` and `update-pump` actions while seems HTTP API only have `pause` and `close` ?

Reply:
yep, what's meaning of `close`? `offline`?

Reply:
`pause` is equivalent to `pause-pump\pause-drainer`, close is equivalent to `offline-pump\offline-drainer`.
the function of `update` are not supported by HTTP API.